src/main/java/app/freerouting/gui/MainApplication.java

```diff
@@ -369,45 +370,78 @@ private void ExportBoardToFile(String filename) {
 
         // Get the correct panel from the option pane
         JPanel optionPanel = null;
+        int startButtonIndex = -1;
+        int cancelButtonIndex = -1;
         for(Component c : optionPane.getComponents())
           // Ensure there are buttons on the panel
-          if(c instanceof JPanel && ((JPanel)c).getComponents()[0] instanceof JButton){
+          if(c instanceof JPanel && ((JPanel)c).getComponents()[0] instanceof JButton && ((JPanel)c).getComponents()[1] instanceof JButton){
             optionPanel = (JPanel)c;
+
+            JButton firstButton = (JButton)optionPanel.getComponents()[0];
+            JButton secondButton = (JButton)optionPanel.getComponents()[1];
+
+            if (firstButton.isDefaultButton())
+            {
+              // we normally have the first button as the default
+              startButtonIndex = 0;
+              cancelButtonIndex = 1;
+            } else if (secondButton.isDefaultButton())
+            {
+              // we might have a UI/localization where the second button is the default
+              startButtonIndex = 1;
+              cancelButtonIndex = 0;
+            }
+
             break;
           }
 
         // Set the default button to "Start now"
-        JButton startButton = (JButton)(optionPanel.getComponents()[0]);
-        startButton.setText(resources.getString("auto_start_routing_startnow_button"));
-        startButton.addActionListener(e -> {
-          ModelDialogTimeout = 0;
-          dialog.dispose();
-        });
+        if ((optionPanel != null) && (startButtonIndex >= 0)) {
+          JButton startButton = (JButton) (optionPanel.getComponents()[startButtonIndex]);
+          startButton.setText(resources.getString("auto_start_routing_startnow_button"));
+          startButton.addActionListener(
+              e -> {
+                ModelDialogTimeout = 0;
+                dialog.dispose();
+              });
+        }
 
         // Set the cancel button to "Cancel (20)"
-        JButton cancelButton = (JButton)(optionPanel.getComponents()[1]);
-        cancelButton.setText(resources.getString("auto_start_routing_cancel_button") + "(20)");
-        cancelButton.addActionListener(e -> {
-          ModelDialogTimeout = -1;
-          dialog.dispose();
-        });
+        JButton cancelButton;
+        if ((optionPanel != null) && (cancelButtonIndex >= 0)) {
```

Comment:
`cancelButton` is declared in name only outside of the if-statement, whereas `startButton` is fully declared. I know that `startButton` is not used in a timer, but for readability it makes sense to have both of these buttons similar so that in the future they are coded in the same way.

Reply:
Good insight!